### PR TITLE
Fix "preset-all" with dangling symlinks and install-section hint emitted too eagerly

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/shared/install.c [hunk 1]
```diff
@@ -116,6 +116,14 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_PATH);
 }
 
+static const char *unitfiletype_table[_UNIT_FILE_TYPE_MAX] = {
+        [UNIT_FILE_TYPE_REGULAR] = "regular",
+        [UNIT_FILE_TYPE_SYMLINK] = "symlink",
+        [UNIT_FILE_TYPE_MASKED] = "masked",
+};
+
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP_TO_STRING(unitfiletype, UnitFileType);
```
Comment (poettering) [inline src/shared/install.c]: Not that it really matters: usually when mapping CamelCase syntax into underscore_separated syntax `FooBar` becomes `foo_bar` in our sources. Hence, call the table `unit_file_type_table`?

**Comment by poettering** [comment]
I agree with dropping the "draw_" prefix from draw_special_char, it doesn't really make sense indeed. However, I#d really prefer if we could maintain some common prefix for the chracter enum, simply because when seeing an upper case string called "ARROW" in C code, I might think it actually is defined as:

``` c
#define ARROW '→'
```

Even thought it really isn't, and the above doesn't even work, and it really is defined to something like:

``` c
#define ARROW 6
```

Maybe we shouldn't call this "char" either btw, since for the tree stuff it really is more than one char...

maybe call it special_symbol() or special_glyphs() or so, and then define the enum as GLYPH_ARROW, GLYPH_DASH, ... or SYMBOL_ARROW, SYMBOL_DASH, …

Not that all that really mattered, but it's you who opened the box of pandora here ;-)

Anyway, I am fine with merging this as is, actually, as nothing of this matters... It's up to you...

**Comment by keszybz** [comment]
Force pushed with the following changes:
- add underscores in "unit_file_type"
- rename special_char() to special_glyph()
- constify special_glyph()

I didn't add the GLYPH_ prefix to the enum values, because it seems to be a bit too much to have "glyph" repeated in `special_glyph(GLYPH_ARROW)`.

**Comment by poettering** [comment]
i still don't like the non-prefixed definition of ARROW and stuff, I really think this should be called GLYPH_ARROW, but OK.

**Comment by poettering** [comment]
the ubunti CI issue appears unrelated (some SSL stuff), will ignore. (@martinpitt please have a look?)

**Comment by martinpitt** [comment]
> the ubunti CI issue appears unrelated (some SSL stuff)

Looks like a temporary network glitch when trying to clone from anonscm.debian.org. I already retry after some delay on failure, but if it fails twice in a row, pretty well the only thing we can do is to manually retry the test after an hour or so. But indeed this can be ignored.

**Pull request merged**